forest/daemon/src/daemon.rs

```diff
@@ -398,18 +391,51 @@ pub(super) async fn start(config: Config, detached: bool) {
     info!("Forest finish shutdown");
 }
 
+async fn prompt_and_fetch_snapshot(config: &mut Config) {
+    if !config.client.download_snapshot {
+        if let false = Confirm::with_theme(&ColorfulTheme::default())
+            .with_prompt(
+                "Forest needs a snapshot to sync with the network. Would you like to download one now?",
+            )
+            .default(false)
+            .interact()
+            .expect("infallible. yes or no with default")
+        {
+            cli_error_and_die(
+                "Forest cannot sync without a snapshot. Download a snapshot from a trusted source and import with --import-snapshot=[file]",
+                1
+            );
+        }
+    }
+
+    let snapshot_path = config
+        .client
+        .data_dir
+        .join("snapshots")
+        .join(config.chain.name.clone());
```

Comment:
Let's merge this with the identical code in `chain_cmd.rs`.

Reply:
extracted it out